content/posts/5-grzechow-juniora.mdx

```diff
@@ -0,0 +1,68 @@
+robienie tasków samemu. Choćby miało to zająć miesiać to junior chce udowodnić, ze on potrafi! No
+i zazwyczaj kończy się to wręcz przeciwnie, co daily dostajesz pytanie czy juz konczysz tego
+taska, czy moze potrzebujesz pomocy, a ty idąc dalej w zaparte mówisz, ze wszystko jest super
+jednocześnie niepotrzebnie zwiększając swój poziom stresu. Zapamiętaj, nie ma nic złego w
+pytaniu o pomoc czy o pair programming! To nie tylko pomoze ci szybciej rozwiazac twojego taska,
+ale przede wszystkim zwiększy poziom twojej wiedzy, którą zdobędziesz od swojego zespołu.
+
+## Próby pokazania swojej wiedzy na temat róznych bibliotek/frameworków
+
+Jedną z paru rzeczy, które odrózniają juz bardziej doświadczonego programiste od juniora jest
+pewne wyczucie, kiedy jakiś task wymaga biblioteki zewnętrznej bo implementacja i pokrycie
+wszystkich case'ow zajmie zbyt duzo czasu. Niestety tego brakuje wielu juniorom i bardzo łatwo
+jest to zauwazyć juz w repozytoriach takich osób. Projekt w React ToDo-Lista, a w paczkach
+doinstalowane Redux, Redux-Saga, Redux Observable, RxJS, które ostatecznie są uzyte do trzymania
+tasków. No cóz po prostu overengineering pełną gęba, ale za to jakie to uczucie, gdy mozna
+napisać, ze uzyło sie X popularnej technologii u siebie w projekcie! Pamiętaj, biblioteki racja są
+po to by ułatwić nam zycie, ale kazda kolejna biblioteka w projekcie to coraz większy próg wejścia
+dla nowych osób, tak więc zastanów sie dwa razy zanim sie pochwalisz wiedzą o jakiejś turbo nowej
+mega bibliotece i spróbujesz wrzucisz ją do projektu.
+
+## Nie wyciaganie wniosków z Code Review
+
+Tak! Tacy juniorzy tez sie niestety zdarzają. Od razu tłumaczę, ze nie chodzi mi o całkowite
+zignorowanie CR i dalsze popełnianie tych samych błędów, a po prostu o przeświadczenie juniora, ze
+rozwiązanie bardziej doświadczonego kolegi jest gorsze niz jego bo zobaczył je u jakiegoś
+wielkiego youtubera. Nikt z nas nie jest nieomylny tak więc nie popadajmy ze skrajności w
+skrajność, z robienia jak youtuber do robienia jak osoba z zespołu. Pamiętaj, kazde Code Review to kolejny
+krok w celu awansu twojej wiedzy. Kazdy komentarz wez sobie głęboko pod analize i staraj sie
+zrozuzmieć, dlaczego X, a nie Y. To jeden z najlepszych sposobów na pozyskanie doswiadczenia i
+wiedzy.
+
+## Brak znajomości narzedzi do debuggowania
+
+Przejdzmy teraz do bardziej technicznego tematu. Na początku drogi kazdy z nas jest skupiony na
+jak najlepszym poznaniu technologii w której pisze. Czy to C# czy JS czy jakikolwiek inny język to
+jednak najczęściej wybieranym sposobem "debuggowania" aplikacji jest CDD czyli Console Driven
+Development. No nie ma co się kłócić w prostych przypadkach jest to dość szybkie rozwiązanie
+jezeli chcemy sprawdzic co sie znajduje w X zmiennej. Jednak w przypadku juz bardzo rozległych
+aplikacji wiedza o tym jak uzywac debuggera jest wymagana! Dlatego nie pomijaj etapu jego nauki,
+gdyz szybko ci ona odda :D
```

Comment:
Nie korzystam z dubbugera XD

Reply:
+1 byczq

Reply:
Niekoniecznie z debuggera, ale warto znać mozliwości devtoolsów, wtyczek do rijakta/apollo/reduksa itp.

I cały punkt bym zmienił na znajomosc calego zestawu narzedzi, ale to akurat nie tylko grzech juniorow xD

wiec artykul by trzeba zmienic na "rady dla juniora"